### PR TITLE
Validate packets before emitting raw data packets

### PR DESCRIPTION
**Pull request by pushtheworldllc**
### Enhancements

* Add tests for parsing raw packets

### Bug Fixes

* Removed `got here` log from `.streamStart()`
* Validate stop byte before emitting `rawDataPacket`

### CODE REVIEW

File: openBCIBoard.js [hunk 1]
```diff
@@ -1149,8 +1148,8 @@ function OpenBCIFactory() {
             while (readingPosition <= bytesToRead - k.OBCIPacketSize) {
                 if (data[readingPosition] === k.OBCIByteStart) {
                     var rawPacket = data.slice(readingPosition, readingPosition + k.OBCIPacketSize);
-                    this.emit('rawDataPacket',rawPacket);
                     if (data[readingPosition + k.OBCIPacketSize - 1] === k.OBCIByteStop) {
+                        this.emit('rawDataPacket',rawPacket);
```
Comment (andrewheusser) [inline openBCIBoard.js]: so now it just makes sure the stop byte is there before emitting the packet? how does this take care of the bug (packet dropped every 30ms)?

Comment (pushtheworldllc) [inline openBCIBoard.js]: This does not fix the bug i think i found. This PR simply prevents packets being emitted from the bug i found. I might make just make another PR with the 1.0.0 packet parsing method in an update called 0.3.10

Comment (andrewheusser) [inline openBCIBoard.js]: ok, otherwise this looks good, want me to merge this?

Comment (pushtheworldllc) [inline openBCIBoard.js]: go for gold!

**Pull request merged**